tests/unit/experimental/pipeline/test_orchestrator.py

```diff
@@ -16,13 +16,76 @@
 from neo4j_genai.experimental.pipeline import Component
 from neo4j_genai.experimental.pipeline.pipeline import Orchestrator, Pipeline, RunStatus
 
+from tests.unit.experimental.pipeline.components import ComponentPassThrough
+
+
+def test_orchestrator_get_component_inputs_from_user_only() -> None:
+    pipe = Pipeline()
+    pipe.add_component(ComponentPassThrough(), "a")
+    pipe.add_component(ComponentPassThrough(), "b")
+    orchestrator = Orchestrator(pipe)
+    data = orchestrator.get_component_inputs("a", {}, {"a": {"value": "text"}})
+    assert data == {"value": "text"}
+    data = orchestrator.get_component_inputs("b", {}, {"a": {"value": "text"}})
+    assert data == {}
+
+
+def test_pipeline_get_component_inputs_from_parent_specific() -> None:
+    pipe = Pipeline()
+    pipe.add_component(ComponentPassThrough(), "a")
+    pipe.add_component(ComponentPassThrough(), "b")
+    pipe.connect("a", "b", input_config={"value": "a.result"})
+
+    # component "a" already run and results stored:
+    pipe._store.add("a", {"result": "a_result"})
```

Comment:
Is it worth renaming `a_result` to something else? I think I'm getting confused why `a_result` should be returned instead of `"text"`

Reply:
Component "a" takes "text" as input and outputs "a_result". Component "b" takes the output of "a" as input, so "a_result" is the input for "b". Does that make sense?
I'm happy with any suggestion for a better name for "a_result" if you think it's less confusing :)

Reply:
I see. Perhaps replace `a_result` with `a_output` and replace `text` with `a_input`?

Reply:
I think if I were to reread this in future, I will get confused between the purpose of `a_result`, `a.result`, and `text`

Reply:
Done, used a longish but meaningful string "output from component XX".